src/Draco.Compiler/Internal/Symbols/Metadata/MetadataPropertySymbol.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Linq;
+using System.Reflection.Metadata;
+
+namespace Draco.Compiler.Internal.Symbols.Metadata;
+
+/// <summary>
+/// Represents properties that are read up from metadata.
+/// </summary>
+internal sealed class MetadataPropertySymbol : PropertySymbol, IMetadataSymbol
+{
+    public override TypeSymbol Type => this.Getter?.ReturnType ?? this.Setter?.Parameters[0].Type ?? throw new InvalidOperationException();
+
+    public override FunctionSymbol? Getter
+    {
+        get
+        {
+            if (this.getterNeedsBuild) this.getter = this.BuildGetter();
+            return this.getter;
+        }
+    }
+    private FunctionSymbol? getter;
+    private bool getterNeedsBuild = true;
+
+    public override FunctionSymbol? Setter
+    {
+        get
+        {
+            if (this.setterNeedsBuild) this.setter = this.BuildSetter();
+            return this.setter;
+        }
+    }
+    private FunctionSymbol? setter;
+    private bool setterNeedsBuild = true;
```

Comment:
Why all this crufty needs build logic with bool? I understand that on null this would re-ask the accessor, but let's assume it's not a problem for now and just do the simpler thing.